components/sup/src/manager.rs

```diff
@@ -961,7 +961,19 @@ impl Manager {
                                       self.feature_flags,
                                       pair.clone());
 
-            self.cleanup_packages(&SUP_PKG_IDENT.parse().unwrap()).await;
+            let fully_qualified_ident =
+                PackageIdent::from_str(&format!("{}/{}", SUP_PKG_IDENT, VERSION)).unwrap();
+            let ident = SUP_PKG_IDENT.parse().unwrap();
+            // Check if we are running the most recent version of the Supervisor. If we are, cleanup
+            // the Supervisor packages. If we are not, it does not make sense to cleanup because we
+            // might uninstall the package that is running.
+            if let Some(package_install) = pkg::installed(&ident) {
+                if fully_qualified_ident.fully_qualified()
```

Comment:
Shouldn't `fully_qualified_ident` always be fully-qualified, by definition?

Actually, `VERSION` only includes the version, not version+release, so I think this will _never_ be fully-qualified.

Reply:
Yeah, that was a bad name. Addressed in [bb9151](https://github.com/habitat-sh/habitat/pull/7582/commits/bb91519cbbf3e1d4b5a73e2ec01242cc21333e0d).

Depending on if the supervisor was built using `hab pkg build` or `cargo build` `VERSION` may or may not produce a fully qualified ident. See [here](https://github.com/habitat-sh/habitat/blob/20243c481238afa46851931ffc8029502a7969b2/components/libbuild.rs#L13). There is probably some cleanup that could happen with how we populate that `VERSION` file.